### PR TITLE
Run AWS ACK code-generator with go.mod --aws-sdk-go-version

### LINKED ISSUE #1414: Version bump of aws-sdk-go results in invalid field type
### What happened?
Since we need some features release with the `aws-sdk-go` in version 1.43.43, we tried to bump the version. This resuls on `make generate` in 

```
...
13:40:12 [ OK ] Generating sfn controllers and CRDs
13:40:12 [ .. ] Generating transfer controllers and CRDs
13:40:35 [ OK ] Generating transfer controllers and CRDs
13:40:35 [ .. ] go generate linux_amd64
/home/pschroeder/repos/provider-aws/apis/ec2/v1alpha1/zz_i_pam.go:55:2: invalid field type: invalid type
/home/pschroeder/repos/provider-aws/apis/ec2/v1alpha1/zz_i_pam_pool.go:107:2: invalid field type: invalid type
/home/pschroeder/repos/provider-aws/apis/ec2/v1alpha1/zz_i_pam_scope.go:50:2: invalid field type: invalid type
/home/pschroeder/repos/provider-aws/apis/ec2/v1alpha1/zz_network_insights_access_scope.go:47:2: invalid field type: invalid type
/home/pschroeder/repos/provider-aws/apis/ec2/v1alpha1/zz_public_i_pv_4_pool.go:42:2: invalid field type: invalid type
/home/pschroeder/repos/provider-aws/apis/ec2/v1alpha1/zz_public_i_pv_4_pool.go:42:2: unknown type CustomPublicIPv4PoolParameters
/home/pschroeder/repos/provider-aws/apis/ec2/v1alpha1/zz_network_insights_access_scope.go:47:2: unknown type CustomNetworkInsightsAccessScopeParameters
/home/pschroeder/repos/provider-aws/apis/ec2/v1alpha1/zz_i_pam_scope.go:50:2: unknown type CustomIPamScopeParameters
/home/pschroeder/repos/provider-aws/apis/ec2/v1alpha1/zz_i_pam_pool.go:107:2: unknown type CustomIPamPoolParameters
/home/pschroeder/repos/provider-aws/apis/ec2/v1alpha1/zz_i_pam.go:55:2: unknown type CustomIPamParameters
/home/pschroeder/repos/provider-aws/apis/ecr/v1alpha1/zz_pull_through_cache_rule.go:43:2: invalid field type: invalid type
/home/pschroeder/repos/provider-aws/apis/ecr/v1alpha1/zz_pull_through_cache_rule.go:43:2: unknown type CustomPullThroughCacheRuleParameters
/home/pschroeder/repos/provider-aws/apis/efs/v1alpha1/zz_replication_configuration.go:41:2: invalid field type: invalid type
/home/pschroeder/repos/provider-aws/apis/efs/v1alpha1/zz_replication_configuration.go:41:2: unknown type CustomReplicationConfigurationParameters
/home/pschroeder/repos/provider-aws/apis/glue/v1alpha1/zz_session.go:68:2: invalid field type: invalid type
/home/pschroeder/repos/provider-aws/apis/glue/v1alpha1/zz_session.go:68:2: unknown type CustomSessionParameters
/home/pschroeder/repos/provider-aws/apis/kafka/v1alpha1/zz_cluster_v_2.go:41:2: invalid field type: invalid type
/home/pschroeder/repos/provider-aws/apis/kafka/v1alpha1/zz_cluster_v_2.go:41:2: unknown type CustomClusterV2Parameters
/home/pschroeder/repos/provider-aws/apis/lambda/v1alpha1/zz_function_url_config.go:57:2: invalid field type: invalid type
/home/pschroeder/repos/provider-aws/apis/lambda/v1alpha1/zz_function_url_config.go:57:2: unknown type CustomFunctionURLConfigParameters
Error: not all generators ran successfully
run `controller-gen object:headerFile=../hack/boilerplate.go.txt paths=./... crd:allowDangerousTypes=true,crdVersions=v1 output:artifacts:config=../package/crds -w` to see all available markers, or `controller-gen object:headerFile=../hack/boilerplate.go.txt paths=./... crd:allowDangerousTypes=true,crdVersions=v1 output:artifacts:config=../package/crds -h` for usage
exit status 1
apis/generate.go:26: running "go": exit status 1
angryjet: error: error loading packages using pattern ./...: /home/pschroeder/repos/provider-aws/apis/eks/v1alpha1/zz_generated.resolvers.go:27:13: cannot use &(v1beta1.ClusterList literal) (value of type *"github.com/crossplane-contrib/provider-aws/apis/eks/v1beta1".ClusterList) as "github.com/crossplane/crossplane-runtime/pkg/resource".ManagedList value in struct literal: missing method GetItems
exit status 1
apis/eks/v1alpha1/custom_types.go:17: running "go": exit status 1
13:42:10 [FAIL]
make[1]: *** [build/makelib/golang.mk:271: go.generate] Error 1
make: *** [build/makelib/common.mk:419: generate] Error 2
```

### How can we reproduce it?

```bash
git clone https://github.com/crossplane-contrib/provider-aws
cd provider-aws
<change the version of github.com/aws/aws-sdk-go in go.mod with your prefered editor to v1.43.43>
go mod tidy
make submodules
make generate
```

### What environment did it happen in?
Crossplane version: 

* Kubernetes version:
```
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.3", GitCommit:"816c97ab8cff8a1c72eccca1026f7820e93e0d25", GitTreeState:"clean", BuildDate:"2022-01-25T21:25:17Z", GoVersion:"go1.17.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21+", GitVersion:"v1.21.13-eks-84b4fe6", GitCommit:"e1318dce57b3e319a2e3fecf343677d1c4d4aa75", GitTreeState:"clean", BuildDate:"2022-06-09T18:22:07Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.23) and server (1.21) exceeds the supported minor version skew of +/-1
```
* Kubernetes distribution: AWS EKS
* OS:
```
NAME="Ubuntu"
VERSION="20.04.4 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.4 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```
* Kernel: 
```
Linux DESKTOP-G9JUA33 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

### PR DESCRIPTION
**Pull request by chlunde**
### Description of your changes

- Compile code-generator to avoid link overhead per service
- Run AWS ACK code-generator with go.mod --aws-sdk-go-version

Fixes #1414
Fixes https://github.com/aws-controllers-k8s/community/issues/1441
    
I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

* No changes w/ make services.all
* Compiling code (not tested) with `go get github.com/aws/aws-sdk-go@v1.43.43 && go mod tidy`. Testing new SDKs will still be the responsibility of the change bumping the SDK

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Review by haarchri — APPROVED** [review]
thanks for taking care ;)

File: Makefile [hunk 1]
```diff
@@ -134,17 +134,19 @@ run: go.build
 
 .PHONY: cobertura manifests submodules fallthrough test-integration run crds.clean
 
+GO_SDK_V1_VERSION := $(shell awk '/github.com\/aws\/aws-sdk-go / { print $$2 }' < go.mod)
```
[the remark this answers is not in the document]
Comment (A-Hilaly) [reply Makefile:137]: Maybe also name it `AWS_SDK_GO_VERSION` or `ACK_SDK_GO_VERSION` to keep it consistent with the flag name?

[the remark this answers is not in the document]
Comment (haarchri) [reply Makefile:137]: @chlunde any updates ? =)

[the remark this answers is not in the document]
Comment (chlunde) [reply Makefile]: Thanks, fixed, sorry for the delay!

**Pull request merged**